dtfs-central-api/src/services/changeStream/changeStreamApi.test.ts

```diff
@@ -0,0 +1,81 @@
+import { Binary, ChangeStreamDocument, ChangeStreamCollModDocument, ObjectId } from 'mongodb';
+import axios from 'axios';
+import { postAuditDetails } from './changeStreamApi.ts';
+
+jest.mock('axios', () => jest.fn(() => Promise.resolve('mockResponse')));
+
+describe('changeStreamApi', () => {
+  describe('postAuditDetails', () => {
+    const mockChangeStreamDocument: ChangeStreamDocument = {
+      _id: '123',
+      collectionUUID: new Binary(Buffer.from('123')),
+      fullDocument: {},
+      operationType: 'insert',
+      ns: {
+        coll: 'mockCollection',
+        db: 'mockDb',
+      },
+      documentKey: {
+        _id: new ObjectId(),
+      },
+    };
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      process.env = { ...process.env, AUDIT_API_URL: 'audit API url', AUDIT_API_USERNAME: 'audit API username', AUDIT_API_PASSWORD: 'audit API password' };
+    });
+
+    it('should call api when called with a valid change stream document', async () => {
+      await postAuditDetails(mockChangeStreamDocument);
+
+      expect(axios).toHaveBeenCalledWith({
+        method: 'post',
+        url: expect.any(String) as string,
+        headers: {
+          Authorization: expect.any(String) as string,
+          'Content-Type': 'application/json',
+          integrationHubCollectionName: 'mockCollection',
+          integrationHubItemId: mockChangeStreamDocument.documentKey._id.toString(),
+          integrationHubProcess: 'dtfs',
+        },
+        data: mockChangeStreamDocument.fullDocument,
+      });
+    });
+
+    it('should not call the audit api if the change document is not of correct type', async () => {
+      const changeStreamCollModDocument: ChangeStreamCollModDocument = {
+        _id: '123',
+        collectionUUID: new Binary(Buffer.from('123')),
+        operationType: 'modify',
+      };
+
+      await postAuditDetails(changeStreamCollModDocument);
+
+      expect(axios).not.toHaveBeenCalled();
+    });
+
+    it('should not call the audit api if the audit API url env var is missing', async () => {
+      delete process.env.AUDIT_API_URL;
+
+      await postAuditDetails(mockChangeStreamDocument);
+
+      expect(axios).not.toHaveBeenCalled();
+    });
+
+    it('should not call the audit api if the audit API password env var is missing', async () => {
+      delete process.env.AUDIT_API_PASSWORD;
+
+      await postAuditDetails(mockChangeStreamDocument);
+
+      expect(axios).not.toHaveBeenCalled();
+    });
+
+    it('should not call the audit api if the audit API username env var is missing', async () => {
+      delete process.env.AUDIT_API_USERNAME;
+
+      await postAuditDetails(mockChangeStreamDocument);
+
+      expect(axios).not.toHaveBeenCalled();
+    });
+  });
```

Comment:
Would this work for brevity?
```suggestion
    const envVarTestCases = [
      {envVarName: 'AUDIT_API_URL'},
      {envVarName: 'AUDIT_API_USERNAME'},
      {envVarName: 'AUDIT_API_PASSWORD'},
    ]
    it.each(envVarTestCases)('should not call the audit api if the %envVarName env var is missing', async ({envVarName}) => {
      delete process.env[envVarName];

      await postAuditDetails(mockChangeStreamDocument);

      expect(axios).not.toHaveBeenCalled();
    })
```

Reply:
Lovely suggestion :D